command.go

```diff
@@ -40,6 +47,23 @@ func Cmd(ctx context.Context, parts ...string) *Command {
 	}
 }
 
+// BashWithOpts appends all the given bash options to the bash command with '-o'. The given parts
+// is then joined together to be executed with 'bash -c'
+//
+// The final command will have the following format: bash -o option-1 -c command. For recommended strict bash options
+// see StrictBashOpts, which has 'pipefail' and 'errexit' options
+func BashWithOpts(ctx context.Context, opts []BashOpt, parts ...string) *Command {
```

Comment:
WDYT about `BashWith`? Then:

```go
// "run bash with bash option pipefail"
run.BashWith(ctx, run.BashOptPipeFail, "echo foobar")
```

Reply:
maybe instead of creating another entrypoint of using `bash`, just chain them together?

```go
run.Bash("echo hi").WithBashOption([]run.BashOpt{run.BashOptPipeFail})
```

Reply:
@michaellzc I like that!

Reply:
The downside there is that such an option wouldn't make much sense on `type Command struct` instantiated from `run.Cmd`, which doesn't necessarily use `bash`

Reply:
@michaellzc @bobheadxi made a good point so I went with his suggestion 😄